apiml-security-common/src/main/java/org/zowe/apiml/security/common/error/AuthExceptionHandler.java

```diff
@@ -48,133 +42,131 @@ public AuthExceptionHandler(
+            handleAuthenticationRequired(requestUri, function, addHeader, ex);
         } else if (ex instanceof BadCredentialsException) {
-            handleBadCredentials(request, response, ex);
+            handleBadCredentials(requestUri, function, ex);
         } else if (ex instanceof AuthenticationCredentialsNotFoundException) {
-            handleAuthenticationCredentialsNotFound(request, response, ex);
+            handleAuthenticationCredentialsNotFound(requestUri, function, ex);
         } else if (ex instanceof AuthMethodNotSupportedException) {
-            handleAuthMethodNotSupported(request, response, ex);
+            handleAuthMethodNotSupported(requestUri, function, ex);
         } else if (ex instanceof TokenNotValidException) {
-            handleTokenNotValid(request, response, ex);
+            handleTokenNotValid(requestUri, function, addHeader, ex);
         } else if (ex instanceof NoMainframeIdentityException nmie) {
-            handleNoMainframeIdentity(request, response, nmie);
+            handleNoMainframeIdentity(requestUri, function, addHeader, nmie);
         } else if (ex instanceof TokenNotProvidedException) {
-            handleTokenNotProvided(request, response, ex);
+            handleTokenNotProvided(requestUri, function, ex);
         } else if (ex instanceof TokenExpireException) {
-            handleTokenExpire(request, response, ex);
+            handleTokenExpire(requestUri, function, ex);
         } else if (ex instanceof TokenFormatNotValidException) {
-            handleTokenFormatException(request, response, ex);
+            handleTokenFormatException(requestUri, function, ex);
         } else if (ex instanceof AccessTokenBodyNotValidException) {
-            handleInvalidAccessTokenBodyException(request, response, ex);
+            handleInvalidAccessTokenBodyException(requestUri, function, ex);
         } else if (ex instanceof InvalidCertificateException) {
-            handleInvalidCertificate(response, ex);
+            handleInvalidCertificate(function, ex);
         } else if (ex instanceof ZosAuthenticationException) {
-            handleZosAuthenticationException(response, (ZosAuthenticationException) ex);
+            handleZosAuthenticationException(function, (ZosAuthenticationException) ex);
         } else if (ex instanceof InvalidTokenTypeException) {
-            handleInvalidTokenTypeException(request, response, ex);
+            handleInvalidTokenTypeException(requestUri, function, ex);
         } else if (ex instanceof AuthenticationException) {
-            handleAuthenticationException(request, response, ex);
+            handleAuthenticationException(requestUri, function, ex);
         } else if (ex instanceof ServiceNotAccessibleException) {
-            handleServiceNotAccessibleException(request, response, ex);
-        } else {
-            throw new ServletException(ex);
+            handleServiceNotAccessibleException(requestUri, function, ex);
         }
```

Comment:
what happens in any other case? There should be some default behaviour, right?

Reply:
handled